spec/cc/yaml/parser/psych_spec.rb

```diff
@@ -7,7 +7,46 @@
       - "*.rb"
       - "test/*"
     YAML
+
     config = CC::Yaml.parse yaml
     config.errors.must_include CC::Yaml::Parser::Psych::NO_ANALYSIS_KEY_FOUND_ERROR
   end
+
+  specify "properly parses engine data" do
+    yaml = <<-YAML
+    engines:
+      duplication:
+        enabled: true
+        config:
+          ruby:
+            mass_threshold: 10
+          javascript:
+            mass_threshold: 20
+    YAML
+
+    parsed_yaml = CC::Yaml.parse yaml
+    config = parsed_yaml["engines"]["duplication"].config
+
+    assert !config.empty?
+    assert_equal config["ruby"]["mass_threshold"], 10
+    assert_equal config["javascript"]["mass_threshold"], 20
+  end
+
+  specify "supports listed values" do
+    yaml = <<-YAML
+    engines:
+      duplication:
+        enabled: true
+        config:
+          languages:
+          - ruby
+          - javascript
+    YAML
+
+    parsed_yaml = CC::Yaml.parse yaml
+    config = parsed_yaml["engines"]["duplication"].config
```

Comment:
We should, as a rule, use methods whenever possible on the config object. So in this case, `parsed_yaml.engines["duplication"].config`.